envi/archs/arm/disasm.py

```diff
@@ -2379,6 +2389,12 @@ def render(self, mcanv, op, idx):
             if self.val & 1<<l:
```

Comment:
these need to come out.  :)
completely duplicating functionality here.

i suggest you test this against a envi.memcanvas.StringMemoryCanvas to test render() functions.  render needs a memcanvas to write to.  StringMemoryCanvas provides one that is just a string which can be checked/compared.  it doesn't actually need a Memory backing (can be None) if using simple functionality like this.
eg.

>>> import envi.memcanvas as ememcanv
>>> emc = ememcanv.StringMemoryCanvas(None)
>>> op.render(emc)
>>> print emc.strval
foo r10, r11